sql/rowexec/join_iters.go

```diff
@@ -306,12 +308,20 @@ func (i *existsIter) Next(ctx *sql.Context) (sql.Row, error) {
 					return nil, err
 				}
 			} else {
+				i.rightIterNonEmpty = true
 				nextState = esCompare
 			}
 		case esRightIterEOF:
 			if i.typ.IsSemi() {
 				// reset iter, no match
 				nextState = esIncLeft
+			} else if !i.rightIterNonEmpty && !isTrueLit(i.cond) {
```

Comment:
This latter condition seems pretty fragile, at least add a note about why a condition would be a literal true (as opposed to an expression that always evaluates to true, like `1`)

And if this is effectively a marker for a certain node type, consider pulling that out and making this logic explicit

Reply:
i tried a couple variations but we would have to create a new sentinel expression mode to do this. It's not possible to manually make a NOT EXISTS with anything but the literal 1, and if someone changes that things will break loudly.